### PR TITLE
fix: migration to fix multiselect translation bug

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3138)
- [x] This change addresses only certain aspects of the issue

## Description
This migration will update all the data for existing applications that were affected by the multiselect question translation bug

## How Can This Be Tested/Reviewed?
After running the migration locally. Try this also with a copy of prod db. 
Once its run, you can run the following:
```
/*
    SELECT
        a.id,
        a.language,
        a.programs,
        a.preferences
    FROM applications a
    WHERE (a.programs != '[]' OR a.preferences != '[]')
        AND a.language != 'en'
        AND a.created_at > '06/01/2022'
    ORDER BY created_at desc
*/
const res = []; // convert the query results to json and slap in this spot

for (const data of res) {
  if (data.programs && data.programs.length) {
    for (const program of data.programs) {
      if (!program.key) {
        console.log(`${data.id} is missing a top level key`);
      }
      if (program.options && program.options.length) {
        for (const option of program.options) {
          if (!option.key) {
            console.log(`${data.id} is missing an option level key`);
          }
        }
      }
    }

    for (const preference of data.preferences) {
      if (!preference.key) {
        console.log(`${data.id} is missing a top level key`);
      }
      if (preference.options && preference.options.length) {
        for (const option of preference.options) {
          if (!option.key) {
            console.log(`${data.id} is missing an option level key`);
          }
        }
      }
    }
  }
}

```
Ideally that should return no results. 

You can also do a spot check of data on partners to verify that existing applications are looking okay

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/migration/1666583981588-multiselectTranslationFix.ts [hunk 1]
```diff
@@ -0,0 +1,1583 @@
...
+    const applications = await queryRunner.query(`
+        SELECT
+            a.id,
+            a.language,
+            a.programs,
+            a.preferences
+        FROM applications a
+        WHERE (a.programs != '[]' OR a.preferences != '[]')
+            AND a.language != 'en'
+            AND a.created_at > '06/01/2022'
```
Comment (ludtkemorgan) [inline backend/core/src/migration/1666583981588-multiselectTranslationFix.ts:1488]: Is this date chosen arbitrarily? I guess the better question is this here do reduce the number of applications it needs to run through?

**Review by ludtkemorgan — COMMENTED** [review]
I pulled down the PR and got the prod HBA data. After running the migration I'm still seeing the CSV columns with the different languages. For example application `5bc04b6e-290a-44c6-af88-b2c158a577f3` still has the preference json in Filipino. Are you seeing that as well?

File: backend/core/src/migration/1666583981588-multiselectTranslationFix.ts (hunk 1 above)
Comment (YazeedLoonat) [reply backend/core/src/migration/1666583981588-multiselectTranslationFix.ts:1490]: okay I think I got this updated correctly

and that date is about 1 month before the multiselect changes went live, so ideally this should catch everything without having to go through actually every single application

**Review by ludtkemorgan — APPROVED** [review]
This seems to be working now!

**Review by seanmalbert — APPROVED** [review]
Looks good, I'm wondering though what the process is for validating that the script worked across the board. Is it to get it up on staging with prod data and have a thorough review? Can we try and get this in for tomorrow's bug bash so we can use that time to QA the outcome.

**Comment by YazeedLoonat** [comment]
Hey @seanmalbert yeah I think the best way is to do a deeper check to make sure the data looks correct against production data

at least in the mean time the script that I had attached should be a good um canary in the coal mine style test 

where if we have data without a key that means something went wrong

**Pull request merged**